chore: Bump Sentry-Cocoa version (#3250)
getsentry/sentry-react-native: CHANGELOG.md

# Changelog

### Dependencies

## Unreleased

- Bump Cocoa SDK from v8.9.4 to v8.10.0 ([#3250](https://github.com/getsentry/sentry-react-native/pull/3250))

>   - This will fix an error to compile projects that use cocoapod with `use_frameworks!` option.

Reviewer: ```suggestion
  - This fixes a compile error for projects that use CocoaPods with `use_frameworks!` option.
```

  - [changelog](https://github.com/getsentry/sentry-cocoa/blob/main/CHANGELOG.md#8100)
  - [diff](https://github.com/getsentry/sentry-cocoa/compare/8.9.4...8.10.0)

## 5.9.0

## Important Note

> **Do not use this version** if you use cocoapod with `use_frameworks!` option. It introduces a bug where the project won't compile.

Reviewer: ```suggestion
**Do not use this version** if you use CocoaPods with `use_frameworks!` option. It introduces a bug where the project won't compile.
```
